### PR TITLE
added login page, top-level login info via context

### PR DESCRIPTION
**Pull request by jiangerr**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import { BrowserRouter, Route, Routes } from "react-router-dom";
 import LandingPage from "./pages/LandingPage/LandingPage";
+import LoginPage from "./pages/LoginPage/LoginPage";
 import "./App.scss"
 import React from "react";
 import StaffRegistration from "./pages/StaffRegistration/StaffRegistration";
 import YouthRegistration from "./pages/YouthRegistration/YouthRegistration";
+import UserContext from "./context/UserContext";
```
Comment (alexbawa) [inline src/App.js:8]: I have never worked with this before - what's it doing, how should it behave? Can you send me documentation

File: src/components/ProgressionButton/ProgressionButton.js [hunk 2]
```diff
@@ -2,25 +2,30 @@ import './ProgressionButton.scss';
 import Forward from "./assets/arrow-forward.png"
 import Backward from "./assets/arrow-back.png";
 import React from "react";
+import { Link } from "react-router-dom";
```
Comment (alexbawa) [inline src/components/ProgressionButton/ProgressionButton.js:5]: These changes defeat the purpose of what this component is for - you can create a new component for navigation but leave this as it is, it's meant for button clicks that lead to something executing

File: src/context/UserContext.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { createContext } from "react";
+
+const UserContext = createContext({});
+export default UserContext;
```
Comment (alexbawa) [inline src/context/UserContext.js:4]: Again idk what this is so fill me in

File: src/pages/LoginPage/LoginPage.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            const user = await Firebase.loginUser(email, password);
+            this.context.onLogin(user); // update user info at the top level
+        } catch (err) {
+            console.log(err);
+        }
+    };
+
+    handleChange = (event) => {
+        event.preventDefault();
+        this.setState({ [event.target.name]: event.target.value });
```
Comment (alexbawa) [inline src/pages/LoginPage/LoginPage.js:30]: what are the array brackets for? I get how the is updating email/password, but the brackets confuse me

**Review by alexbawa — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (jiangerr) [reply src/App.js:8]: https://reactjs.org/docs/context.html#when-to-use-context

Let me know if there's a better/different way you'd like to approach this implementation

File: src/pages/LoginPage/LoginPage.js (hunk 4 above)
Comment (jiangerr) [reply src/pages/LoginPage/LoginPage.js:30]: I believe event.target.name basically acts as a key in a key-value pair, so it needs to be bracket-enclosed.

File: src/App.js (hunk 1 above)
Comment (jiangerr) [reply src/App.js:8]: From what I understand, contexts are useful for properties that need to be accessible across all levels of the application. This pattern seemed like a good fit for user login data, and I haven't implemented this sort of thing before so I'm unfamiliar with the typical solution if use of contexts is uncommon

Comment (alexbawa) [reply src/App.js:8]: nah after reading that totally makes sense I'm cool with it just wasn't familiar.

File: src/pages/LoginPage/LoginPage.js (hunk 4 above)
Comment (alexbawa) [reply src/pages/LoginPage/LoginPage.js:30]: If it's a key would you need to put the brackets next to the pair you are indexing though, like `whateverPairName[event.target.name]`? Lone brackets to me insinuate creating an array

Comment (alexbawa) [reply src/pages/LoginPage/LoginPage.js:30]: If you can test this and show me its working I'll chill just don't trust inherently sorry g

Comment (jiangerr) [reply src/pages/LoginPage/LoginPage.js:30]: Screenshot 2023-01-03 at 5 16 59 AM

right above Firebase error - logging state to console right before it's used by Firebase.loginUser, email and password values are being assigned correctly

Comment (alexbawa) [reply src/pages/LoginPage/LoginPage.js:31]: you right

**Review by JhanaviThakkar — APPROVED** [review]

**Review by alexbawa — APPROVED** [review]

**Pull request merged**